backend/app/finders/spree/admin/reports/orders/top_products_by_line_item_totals.rb

```diff
@@ -0,0 +1,57 @@
+module Spree
+  module Admin
+    module Reports
+      module Orders
+        class TopProductsByLineItemTotals < Base
+          def initialize(params, opts = {})
+            @params = params
+            @currency = opts[:currency]
+          end
+
+          def call
+            variants = Spree::Variant.joins(:prices, line_items: :order)
+            variants = by_completed_at_min(variants)
+            variants = by_completed_at_max(variants)
+
+            variants = variants.group('spree_variants.sku')
+                               .select('spree_variants.sku, sum(spree_line_items.quantity) * spree_prices.amount as line_item_total')
+                               .where('spree_prices.currency = ?', currency)
+                               .order(line_item_total: :desc)
+
+            variants = by_top(variants)
+
+            variants.to_a.map { |v| [v.sku, v.line_item_total] }
+          end
+
+          private
+
+          attr_reader :currency
+
+          def top
+            return nil if params[:top].nil?
+
+            params[:top].to_i
+          end
+
+          def by_top(variants)
+            return variants unless top
```

Comment:
@mbajur when method `top` will return `true` or `false` ?

Reply:
@piotrleniec-spark i think that was your change when you were working on POC for SQL-based solution. Do you remember the reason?

Reply:
![image](https://user-images.githubusercontent.com/581569/70711880-cca87480-1ce2-11ea-9f0d-637887b23356.png)

never, it's just a shorthand notation for `top.present?`